### DISCUSSION TITLE
Shelly 2.5 e firmware alternativi

### DISCUSSION THREAD

**Discussion by cl-7k-01** [indomus/forum#2271] [Hardware & Firmware]
Chiedo venia se la domanda sembrerà banale e scontata ma.. Spesso viene consigliato di "flasshare" firmware alternativi sui dispositivi Shelly e, soprattutto, sui Sonoff.
Nel caso del Shelly 2.5 da usare in configurazione Roller Shutter, con un firmware alternativo non perdo tutte quelle importantissime e imprescindibili impostazioni per l'uso corretto di una tapparella motorizzata, come per esempio l'Open/Close Working Time, Positioning controls per la calibrazione (e quindi la posizione), Safety per arrestare in caso di sforzi ecc. ecc.?
In definitiva: per l'uso su tapparelle elettrica, esiste una vera e valida alternativa allo Shelly 2.5 e al suo firmware proprietario?

**Comment by snakuzzo** [comment]
No...conviene restare con il fw proprietario.

**Reply by cl-7k-01** [reply]
Speravo in una risposta diversa lo ammetto...

**Reply by KC-inDomus** [reply]
perché "speravi"?

**Reply by snakuzzo** [reply]
la cosa migliore che c'è per le tapparelle è proprio il firmware di Shelly....e tu dici "speravo in una risposta diversa" ? :D :D

**Reply by KC-inDomus** [reply]
sì infatti

**Reply by cl-7k-01** [reply]
Semplicemente speravo di sbagliarmi :)
Speravo ci fosse un prodotto altrettanto valido con un firmware all'altezza. Un prodotto di cui ignoravo l'esistenza. Oppure che almeno mi sbagliassi sulle funzioni dei firmware alternativi. Della serie: "Su Tasmota (o ESPHome) ti sbagli, ha tutte le funzioni che cerchi per gestire in sicurezza una tapparella...".
Tutto qui, non so se mi sono spiegato.
E aggiungo: speriamo che nel prossimo futuro ci sia un po' di sana concorrenza in tutti i livelli altrimenti se domattina Shelly aumentasse di botto il prezzo del suo prodotto saremo nei guai...

**Reply by snakuzzo** [reply]
ma guarda che Tasmota ed Esphome sostanzialmente ti offrono tutte le possibilità che hai con il firmware Shelly.
Una cosa che manca è l'autocalibrazione che, se sei in grado, almeno con esphome puoi fartela da solo

**Reply by cl-7k-01** [reply]
> ma guarda che Tasmota ed Esphome sostanzialmente ti offrono tutte le possibilità che hai con il firmware Shelly. Una cosa che manca è l'autocalibrazione che, se sei in grado, almeno con esphome puoi fartela da solo

Forse ho posto male la domanda all'inizio, ma chiedevo proprio questo. ESPHome quindi secondo te è una valida alternativa al firmware Shelly? La calibrazione mi sembra di capire sia fattibile, e le altre funzioni ci sono?
Nel firmware Shelly per esempio è interessante la funzione che ferma/inverte (se non ricordo male) la tapparelle in caso di "sforzo" eccessivo: a chi non è capitato di lasciare qualcosa in traiettoria della tapparella con questa che ci va sopra..
Purtroppo non conosco e non ho avuto modo di provare ne ESPHome ne Tasmota.

**Reply by snakuzzo** [reply]
Ma infatti secondo me stai partendo dal presupposto sbagliato che firmware alternativi abbiano già le funzioni che cerchi e che siano solo da abilitare. Se così fosse hai proprio capito male...Queste funzionalità ed i relativi comportamenti te le devi scrivere da solo!
Ecco perché ti dico, non ha senso sugli Shelly cambiare il firmware (salvo esigenze specifiche). Il firmware di Shelly è completissimo così e nella stragrande maggioranza dei casi non c'è alcun bisogno di cambiare il firmware.

**Reply by cl-7k-01** [reply]
Scusami ma con Tasmota, per esempio, _"fino all’avvento della versione 6.7 di Tasmota tali funzionalità di gestione e controllo tapparelle non erano disponibili: per ottenerle, si faceva ricorso a un fork [omissis] Oggi, invece, ha senso utilizzare direttamente la versione ufficiale del firmware anziché la fork Stefan Bode, oppure migrare da essa alla ufficiale"_, preso da un articolo su InDomus. Per Tasmota quindi non è vero che, almeno per questa specifica funzione, debba scrivere o crearmela da solo: è stata "inglobata" dalla versione 6.7 in poi o sbaglio? Almeno per questa funzione si tratta solamente di "abilitarla"...
Che il firmware di Shelly sia completissimo è un dato di fatto che non ho mai messo in dubbio. Ho solo chiesto se esiste un firmware altrettanto completo, sperando esistesse!
Ho chiesto inoltre se ESPHome o Tasmota, su tutti visto la grande popolarità, disponessero già di queste funzioni (da quel che ho capito per es. Tasmota una l'ha acquisita dalla versione 6.7 in poi...).
"Scriverle" di mio pugno, purtroppo non è, al momento, alla mia portata.

**Reply by snakuzzo** [reply]
e ti sto dicendo che gli altri firmware, così come sono e salvo personalizzazioni, non sono completi come quello nativo Shelly.
E visto che non sai scrivere di tuo pugno, evita il cambio firmware.
Discorso chiuso, credo :)

**Reply by cl-7k-01** [reply]
A costo di sembrare antipatico: vi siete sorpresi della mia "_speravo in una risposta diversa_".
Vi ho solo spiegato il motivo: non mi sembra poi così brutto sperare esista un firmware libero che, così com'è, sia altrettanto valido e paragonabile al firmware Shelly, sbaglio???

 _"non ho chiuso il discorso"_  subito, per questo intervento:

> ma guarda che Tasmota ed Esphome sostanzialmente ti offrono tutte le possibilità che hai con il firmware Shelly. Una cosa che manca è l'autocalibrazione che, se sei in grado, almeno con esphome puoi fartela da solo

Scritto così, per uno che dichiaratamente non conosce Tasmota e ESPHome, sembrava ci fosse già tutto e l'unica cosa da scrivere fosse l'autocalibrazione. Ma poi mi hai chiarito che così non è, almeno non per tutte le funzioni.

Ciao e grazie per aver risposto alle mie domande.

**Reply by indomus** [reply]
ma per carità. hai ragione, non a caso usiamo tutti firmware custom per sentirci più liberi e avere più spazio di manovra. questo però non toglie che una volta tanto quello stock sia ottimo
